src/app/views/apps/modals/app-settings-modal.coffee

```diff
@@ -73,4 +103,117 @@ angular.module 'mnoEnterpriseAngular'
+      {
+        scope:
+          input: ''
+          changeInput: (input) ->
+            if input.length > 2
+              $scope.loadIdMaps($scope.idMaps.entityName, input)
+        template: htmlTemplate
+      }
+
+    optionsForEntities = (data) ->
+      htmlTemplate = '<select ng-init="model=types[getIndexFromValue(model, types)]" ng-model="model" ng-change="changeEntity(model)" ng-options="option for option in types"></select>'
+      {
+        scope:
+          model: data[0]
+          types: data[1]
+          changeEntity: (model) ->
+            $scope.loadIdMaps(model)
+          getIndexFromValue: (value, array) ->
+            _.indexOf(array ,value)
+
+        template: htmlTemplate
+      }
+
+    $scope.loadIdMaps = (externalName, input = null) ->
+      $scope.idMaps.loading = true
+      $scope.idMaps.entityName = externalName
+      params = {
+        per_page: $scope.idMaps.baseSize,
+        page: $scope.idMaps.page,
+        sort: $scope.idMaps.sort
+      }
+      params['filter[name.like]'] = ('%' + input + '%') if input
+      params['filter[external_entity.in]'] = externalName unless externalName == "All"
+      MnoeAppInstances.getIdMaps($scope.app, params)
+        .then((response) ->
+          $scope.mnoIdMapsSubHeaders = [
+            {render: searchInput},
+            {render: optionsForEntities, data: [$scope.idMaps.entityName, $scope.app.addon_organization.entities_types]},
+            {},
+            {}
+          ] unless input
+          $scope.idMaps.list = response.data.map (e) ->
+            att = e.attributes
+            att.connec_id = if att.connec_id then '&#9989;' else '&#10060;'
+            att.external_id = if att.external_id then '&#9989;' else '&#10060;'
+            att
+          $scope.idMaps.totalItems = response.headers('x-total-count')
+          $scope.idMaps.loading = false
+        )
+
+    $scope.disconnect = ->
+      $scope.isDisconnecting = true
+      MnoeAppInstances.disconnect(app)
+        .then((response) ->
+          app.addon_organization.has_account_linked = false
+          app.addon_organization.sync_enabled = false
+          $uibModalInstance.close()
+          toastr.success("Your application has been disconnected")
+          $scope.isDisconnecting = false
+        )
+
+    $scope.sortableSyncsServerPipe = (tableState)->
+      $scope.syncs.sort = updateTableSort(tableState.sort, $scope.syncs.sort)
+      $scope.loadSyncs()
+
+    $scope.sortableIdMapsServerPipe = (tableState)->
+      $scope.idMaps.sort = updateTableSort(tableState.sort, $scope.idMaps.sort)
+      $scope.loadIdMaps($scope.idMaps.entityName)
+
+    updateTableSort = (sortState = {}, sort) ->
+      if sortState.predicate
+        sort = sortState.predicate
+        sort = "-#{sort}" if sortState.reverse
+      return sort
+
+    $scope.selectHistory = ->
+      $scope.isLeftFooterButtonShown = true
+      $scope.selectedTab = 0
+      $scope.textForFooterButton = "Synchronize"
```

Comment:
This is a rendering problematic. This should be in the template, using ng-show on buttons.